### PR TITLE
fix: Using mapInPandas for both spark stream and batch ingestions

### PR DESCRIPTION
**Pull request by EXPEbdodla**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

1. ToPandas() not supporting boolean types so switching to mapInPandas() for stream
2. Feast made updates to Spark Materialization Engine to use mapInPandas already. So making that change to make it consistent with Feast

### CODE REVIEW

File: sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py [hunk 1]
```diff
@@ -184,23 +179,13 @@ def _materialize_one(
...
 
-            # Calculate batch_size per feature_view
-            feature_view_batch_size = (
-                self.repo_config.batch_engine.batch_size
-                if "batch_size" not in feature_view.tags
-                or feature_view.tags["batch_size"] is None
-                else int(feature_view.tags["batch_size"])
-            )
-
             print(
```
Comment (acevedosharp) [inline sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py:182]: Would be nice to eventually use logging.

File: sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py [hunk 2]
```diff
@@ -291,24 +273,7 @@ def write_to_online_store_in_batches(batch_df: pd.DataFrame):
...
-    if "fs_batch" in pandas_dataframe.columns:
-        raise ValueError(
-            "Column 'fs_batch' is reserved by Feature Store. Please rename to avoid conflicts."
-        )
-    pandas_dataframe["fs_batch"] = np.arange(len(pandas_dataframe)) // batch_size
-    pandas_dataframe.groupby("fs_batch").apply(write_to_online_store_in_batches)
-    print(
-        f"INFO!!! Processed partition {pandas_dataframe.shape[0]} records, batch size {batch_size}, time {int((time.time() - start_time))} Seconds"
-    )
+    yield pd.DataFrame([pd.Series(range(1, 2))])  # dummy result because
```
Comment (acevedosharp) [inline sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py:279]:
```suggestion
    yield pd.DataFrame([pd.Series(range(1, 2))])  # dummy result
```

**Review by acevedosharp — APPROVED** [review]

File: sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py (hunk 1 above)
Comment (EXPEbdodla) [reply sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py:182]: Python logging doesn't work directly. We need to fix this at some point.

File: sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py (hunk 2 above)
Comment (EXPEbdodla) [reply sdk/python/feast/infra/materialization/contrib/spark/spark_materialization_engine.py]: Fixed it.

**Pull request merged**